### ISSUE TITLE
Discussion about this idea

### ISSUE THREAD

**Issue by Bigwode** [june6423/Angular-KD#1]
In my opinion, there is insufficient evidence to support that adding noise perturbations solely to the teacher model's logits can effectively improve distillation performance. Would you be willing to provide further discussion or share more common-sense perspectives on this topic?

`def randomize(x):
    x_noise = x.clone()
    noise = torch.randn_like(x_noise)
    x_noise = 0.9 * x_noise + 0.1 * noise
    return x_noise`

**Comment by june6423** [comment]
Thank you for your question and for engaging with our work.

The idea of adding noise perturbations to the teacher's logits was not proposed in our paper; it was originally introduced in the TeKAP paper.

We share your perspective and agree that simply adding noise is often insufficient for effectively improving distillation. Addressing this specific limitation was one of the key motivations for our paper.

For a detailed explanation and the original authors' rationale for that specific method, we recommend referring to the TeKAP paper directly: https://openreview.net/forum?id=DmEHmZ89iB
